### PR TITLE
Store the correct `DGmodel` in `solver_config`

### PR DESCRIPTION
**Pull request by kpamnany**
# Description

For Multirate solvers, the `DGModel` being stored in `solver_config` was not the one being updated.

I have

- [ ] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [ ] Followed all necessary [style guidelines](https://climate-machine.github.io/CLIMA/latest/CodingConventions.html) and run `julia .dev/format.jl`
- [ ] Updated the documentation to reflect changes from this PR.

# For review by CLIMA developers

- [X] There are no open pull requests for this already
- [X] CLIMA developers with relevant expertise have been assigned to review this submission
- [ ] The code conforms to the [style guidelines](https://climate-machine.github.io/CLIMA/latest/CodingConventions.html) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [X] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

**Comment by kpamnany** [comment]
This won't work. `RemainderModel has no field param_set`.

**Comment by kpamnany** [comment]
For diagnostics, [this code](https://github.com/climate-machine/CLIMA/blob/master/src/Diagnostics/atmos_default.jl#L76-L78) uses the diffusive flux which it gets from the `DGModel`. When an experiment was switched to using Multirate, this diagnostic variable immediately went to 0.

File: src/Driver/solver_configs.jl [hunk 1]
```diff
@@ -180,18 +182,18 @@ function SolverConfiguration(
...
-        slow_dg = DGModel(
+        dg = DGModel(
             slow_model,
             grid,
             numerical_flux_first_order,
             numerical_flux_second_order,
             numerical_flux_gradient,
-            state_auxiliary = dg.state_auxiliary,
+            state_auxiliary = state_auxiliary,
         )
```
Comment (jkozdon) [inline src/Driver/solver_configs.jl:195]: Would this work?

```julia
dg = DGModel(
    balance_law,
    grid,
    numerical_flux_first_order,
    numerical_flux_second_order,
    numerical_flux_gradient,
    state_auxiliary = state_auxiliary,
)
slow dg = DGModel(
    slow_model,
    grid,
    numerical_flux_first_order,
    numerical_flux_second_order,
    numerical_flux_gradient,
    state_auxiliary = dg.state_auxiliary,
    state_gradient_flux = dg.state_gradient_flux,
    states_higher_order = dg.states_higher_order,
)
```

Comment (kpamnany) [reply src/Driver/solver_configs.jl:195]: That would be much neater!

Comment (kpamnany) [reply src/Driver/solver_configs.jl:195]: It appears that `fast_dg` is also using `dg.state_auxiliary`. Should these parameters (for `state_gradient_flux` and `states_higher_order` be added there as well? And if so, then should they also be applied to the vertical `DGModel` created for IMEX solvers?

Comment (jkozdon) [reply src/Driver/solver_configs.jl:195]: possibly, though in current use cases the fast (linear) models do not have diffusion so it should have no impact.

**Pull request merged**